### PR TITLE
finished challenge

### PR DESCRIPTION
**Pull request by EllaSvete**

### CODE REVIEW

**Comment by EllaSvete** [comment]
- [ X] Top-level README “Table of Contents” is updated
 - [ X] README for this challenge is complete
       - [ X] Summary, Description, Approach & Efficiency, Solution
       - [X ] Picture of whiteboard
       - [ X] Link to code
 - [X ] Feature tasks for this challenge are completed
 - [X ] Unit tests written and passing
       - [ X] “Happy Path” - Expected outcome
       - [ X] Expected failure
       - [ X] Edge Case (if applicable/obvious)

File: python/data_structures/linked_list.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from email import header
```
Comment (jstreifel-33) [inline python/data_structures/linked_list.py:1]: Looks like your editor may have auto-imported this on accident. I have the same problem and don't really know a great fix. Just have to keep an eye out.

File: python/data_structures/linked_list.py [hunk 2]
```diff
@@ -37,6 +42,65 @@ def includes(self, value):
...
+        new_node = Node(new_value)
+        # fine the node we want to insert before
+        # save value from previous node and check next value if next node is not before value, reassign temp
+        # repeat
+        # if next value is not before then don't reassign temp
+        current = self.head
+        if self.head is None:
+            raise TargetError
+        if self.includes(value) is False:
+            raise TargetError
```
Comment (jstreifel-33) [inline python/data_structures/linked_list.py:78]: While I understand the logic of two conditionals here, consider how includes behaves when the head node is None. Since the result is the same, could you accomplish this with only 1 condition?

File: python/data_structures/linked_list.py [hunk 3]
```diff
@@ -37,6 +42,65 @@ def includes(self, value):
...
+        if current.value is value:
+            new_node.next = self.head
+            self.head = new_node
```
Comment (jstreifel-33) [inline python/data_structures/linked_list.py:81]: lines 79 - 81 match 83 - 85 word for word. In Coding, try to abide by the DRY principle, which stands for Don't Repeat Yourself.

Would the code still function properly if 79 - 81 were removed completely?
